app/src/main/java/lt/vilnius/tvarkau/NewProblemActivity.java

```diff
@@ -230,8 +236,25 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                     break;
                 case REQUEST_PLACE_PICKER:
                     Place place = PlacePicker.getPlace(this, data);
-                    mAddProblemLocation.setText(place.getName());
-                    locationCords = place.getLatLng();
+                    LatLng latLng = place.getLatLng();
+
+                    Geocoder geocoder = new Geocoder(getApplication(), Locale.getDefault());
+
+                    List<Address> addresses = null;
+                    try {
+                        addresses = geocoder.getFromLocation(latLng.latitude, latLng.longitude, 1);
+                    } catch (IOException e) {
+                        e.printStackTrace();
```

Comment:
There weren't any Firebase logging on the first commit, added them later on. So now it's fine.